File: Python/119.py
class Solution:
    # @param {integer} rowIndex
    # @return {integer[]}
    """ Version 1
    def getRow(self, rowIndex):
        curr = [1]
        for i in range(1,rowIndex+1):
            prev = curr
            curr = [1]
            for j in range(1, i):
                curr.append(prev[j-1] + prev[j])
            curr.append(1)
        return curr
    """
    # Version2
    def getRow(self, rowIndex):
        curr = [1]
        for i in range(1, rowIndex+1):
            curr.append(curr[-1]*(rowIndex+1-i)/i)
        return curr
        

File: C++/119.cpp
class Solution {
public:
    /* Version 1
    vector<int> getRow(int rowIndex) {
        vector<int> res;
        res.push_back(1);
        while(rowIndex--){
            vector<int> temp = res;
            for(auto i=1; i<temp.size(); i++)
                res[i] = temp[i-1] + temp[i];
            res.push_back(1);
        }
        return res;
    }
    */
    
    // Version 2 
    vector<int> getRow(int rowIndex) {
        vector<int> res(rowIndex + 1, 0);
        res[0] = 1;
        for(auto i = 1; i < rowIndex + 1; i++)
            for(auto j = i; j > 0; j--) // important!!
                res[j] += res[j-1];
        return res;
    }
};